extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -806,27 +821,53 @@ private OpenAPI generateAnnotationModel(IndexView indexView, Capabilities capabi
         return results;
     }
 
-    private void addStaticModelIfExist(List<Result> results, String path) {
+    private void addStaticModelIfExist(List<Result> results, List<Pattern> ignorePatterns, String path) {
         if (path.endsWith(".json")) {
             // Scan a specific json file
-            addStaticModelIfExist(results, Format.JSON, path);
+            addStaticModelIfExist(results, ignorePatterns, Format.JSON, path);
         } else if (path.endsWith(".yaml") || path.endsWith(".yml")) {
             // Scan a specific yaml file
-            addStaticModelIfExist(results, Format.YAML, path);
+            addStaticModelIfExist(results, ignorePatterns, Format.YAML, path);
         }
     }
 
-    private void addStaticModelIfExist(List<Result> results, Format format, String path) {
+    private void addStaticModelIfExist(List<Result> results, List<Pattern> ignorePatterns, Format format, String path) {
         ClassLoader cl = Thread.currentThread().getContextClassLoader();
-        try (InputStream inputStream = cl.getResourceAsStream(path)) {
-            if (inputStream != null) {
-                results.add(new Result(format, inputStream));
+
+        try {
+            Enumeration<URL> urls = cl.getResources(path);
+            while (urls.hasMoreElements()) {
+                URL url = urls.nextElement();
+                // Check if we should ignore
+                if (!shouldIgnore(ignorePatterns, url.toString())) {
+                    // Add as static model
+                    try (InputStream inputStream = url.openStream()) {
+                        if (inputStream != null) {
+                            byte[] contents = IoUtil.readBytes(inputStream);
+
+                            results.add(new Result(format, new ByteArrayInputStream(contents)));
+                        }
+                    } catch (IOException ex) {
+                        ex.printStackTrace();
```

Comment:
Better rethrow this exception (or `Log.error`) 

```suggestion
                        throw new UncheckedIOException(ex);
```